front50-core/src/main/groovy/com/netflix/spinnaker/front50/ServiceAccountsService.java

```diff
@@ -0,0 +1,146 @@
+import com.netflix.spinnaker.front50.exception.NotFoundException;
+import com.netflix.spinnaker.front50.model.serviceaccount.ServiceAccount;
+import com.netflix.spinnaker.front50.model.serviceaccount.ServiceAccountDAO;
+import com.netflix.spinnaker.kork.exceptions.SystemException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import retrofit.RetrofitError;
+
+@Component
+public class ServiceAccountsService {
+  private static final Logger log = LoggerFactory.getLogger(ServiceAccountsService.class);
+  private static final String MANAGED_SERVICE_ACCOUNT_SUFFIX = "@managed-service-account";
+
+  private final Optional<ServiceAccountDAO> serviceAccountDAO;
+  private final Optional<FiatService> fiatService;
+  private final FiatClientConfigurationProperties fiatClientConfigurationProperties;
+  private final FiatConfigurationProperties fiatConfigurationProperties;
+  private final FiatPermissionEvaluator fiatPermissionEvaluator;
+
+  public ServiceAccountsService(
+      Optional<ServiceAccountDAO> serviceAccountDAO,
+      Optional<FiatService> fiatService,
+      FiatClientConfigurationProperties fiatClientConfigurationProperties,
+      FiatConfigurationProperties fiatConfigurationProperties,
+      FiatPermissionEvaluator fiatPermissionEvaluator) {
+    this.serviceAccountDAO = serviceAccountDAO;
+    this.fiatService = fiatService;
+    this.fiatClientConfigurationProperties = fiatClientConfigurationProperties;
+    this.fiatConfigurationProperties = fiatConfigurationProperties;
+    this.fiatPermissionEvaluator = fiatPermissionEvaluator;
+  }
+
+  Collection<ServiceAccount> getAllServiceAccounts() {
+    return serviceAccountDAO().all();
+  }
+
+  ServiceAccount createServiceAccount(ServiceAccount serviceAccount) {
+    ServiceAccount acct = serviceAccountDAO().create(serviceAccount.getId(), serviceAccount);
+    syncUsers(Arrays.asList(acct));
+    return acct;
+  }
+
+  public void deleteServiceAccount(String serviceAccountId) {
+    ServiceAccount acct = serviceAccountDAO().findById(serviceAccountId);
+    deleteServiceAccounts(Arrays.asList(acct));
+  }
+
+  public void deleteServiceAccounts(Collection<ServiceAccount> serviceAccountsToDelete) {
+    serviceAccountsToDelete.stream()
+        .forEach(
+            sa -> {
+              serviceAccountDAO().delete(sa.getId());
+              try {
+                if (fiatService.isPresent()) {
+                  fiatService.get().logoutUser(sa.getId());
+                }
+              } catch (RetrofitError re) {
+                log.warn("Could not delete service account user $serviceAccountId", re);
```

Comment:
Java doesn't have string interpolation, I assume you mean something like:
```suggestion
                log.warn("Could not delete service account user {}", sa.getId(), re);
```